File: App/velsik/lib/models/apv.dart
import 'question.dart';
import 'user.dart';

class Apv {
  int? apvId;
  int? companyId;
  DateTime? startDate;
  DateTime? endDate;
  List<Question>? questions;
  List<User>? users;

  Apv(
    this.apvId,
    this.companyId,
    this.startDate,
    this.endDate,
    this.questions,
    this.users
  );
  
  // Convert Apv object to a JSON-encodable map
  Map<String, dynamic> toJson() {
    return {
      'apv_id': apvId,
      'company_id': companyId,
      'start_date': startDate?.toIso8601String(), // Convert DateTime to ISO 8601 format
      'end_date': endDate?.toIso8601String(), // Convert DateTime to ISO 8601 format
      'questions': questions?.map((question) => question.toJson()).toList() ?? [], // Convert list of Question objects to JSON
      'users': users?.map((user) => user.toJson()).toList() ?? [], // Convert list of User objects to JSON
    };
  }

}

File: API/VelsikAPI/Database/Models/APV.py

class APV:
    def __init__(self, apv_id, company_id, start_date, end_date, questions, users):
        self.apv_id = apv_id
        self.company_id = company_id
        self.start_date = start_date
        self.end_date = end_date
        self.questions = questions
        self.users = users

    def __str__(self):
        return f"Apv(apv_id={self.apv_id}, company_id={self.company_id}, " \
               f"start_date={self.start_date}, end_date={self.end_date}, " \
               f"questions={self.questions}, users={self.users})"
